### PR TITLE
Python3 update

### PR DESCRIPTION
**Pull request by dlobato**
Initial changes to port ztpserver to python3
#372

### CODE REVIEW

File: client/bootstrap [hunk 1]
```diff
@@ -1334,36 +1344,38 @@ def execute_action(server, action_details, special_attr):
...
-                     if 'attributes' in action_details \
-                     else []
-        try:
-            module.main(Attributes(local_attr, special_attr))
-        except Exception as exc:
-            raise ZtpActionError(exc)
+            if 'attributes' in action_details \
+            else []
+
+        module.main(Attributes(local_attr, special_attr))
```
Comment (mharista) [inline client/bootstrap:1372]: why remove the try/except block here?

File: client/bootstrap [hunk 2]
```diff
@@ -36,7 +36,8 @@
 
 import argparse
 import datetime
-import imp
+import importlib.machinery
+import importlib.util
 import json
 import jsonrpclib
```
Comment (mharista) [inline client/bootstrap:42]: Do we still need to address replacing this and sleekxmpp based on this output from the email thread?

ZeroTouch provisioning bootstrap scripts using 'import jsonrpclib' and
'import sleekxmpp' in python2 will fail. Use python3 instead.

**Review by mharista — CHANGES_REQUESTED** [review]
First round of review

File: setup.py [hunk 3]
```diff
@@ -147,12 +149,11 @@ def ensure_dir(f):
...
     name='ztpserver',
     version=version,
-    description = 'ZTP Server for EOS',
+    description='ZTP Server for EOS',
     author='Arista Networks',
     author_email='eosplus-dev@arista.com',
     url='https://github.com/arista-eosplus/ztpserver',
-    download_url='https://github.com/arista-eosplus/ztpserver/tarball/v%s' % \
-                  version,
+    download_url='https://github.com/arista-eosplus/ztpserver/tarball/v%s' % version,
```
Comment (mharista) [inline setup.py:156]: example to update to f strings.

File: client/bootstrap (hunk 1 above)
Comment (dlobato) [reply client/bootstrap:1345]: block is embedded into another try/except that will raise the exception again into ZtpActionError.
So it looks to me that we are just masking the original exception with no real use.

Comment (mharista) [reply client/bootstrap:1345]: Gotcha.

File: .gitignore [hunk 4]
```diff
@@ -53,3 +53,6 @@ rpm/rpms
 
 # virtualenv
 venv
+
+# IDE
+.idea
```
[the remark this answers is not in the document]
Comment (dlobato) [reply .gitignore]: done

**Review by mharista — CHANGES_REQUESTED** [review]

File: .pylintrc [hunk 5]
```diff
@@ -1,39 +1,28 @@
...
-disable=I0011,C0111,R0921,R0801
+disable=W0703,
+  R0912, # Too many branches
+  R0911, # Too many return statements
+  W0702, # No exception type specified
+  RP0101, # Disable statistics by type
+  I0011, # Allow local disables
+  C0111, # Missing docstring
+  R0801, # Similar lines in 2 files
+  C0209 # consider-using-f-string
```
[the remark this answers is not in the document]
Comment (dlobato) [reply .pylintrc]: Good catch! I missed that one. After removing it, pylint found most of the remaining f-string conversions that where missing

File: actions/copy_file [hunk 6]
```diff
@@ -88,96 +94,77 @@ def main(attributes):
...
-                         'to %s' % (src_url, file_path))
+            node.log_msg(f"copy_file: saving {src_url} to {file_path}")
         except Exception as exc:
-            raise Exception('Unable to retrieve file from URL (%s)' % 
-                            exc)
+            raise RuntimeError("Unable to retrieve file from URL") from exc
 
-        node.log_msg('copy_file: adding rc.eos lines: \n%s$' %
-                     '\n'.join(lines))
+        node.log_msg("copy_file: adding rc.eos lines: \n%s$" % "\n".join(lines))
```
[the remark this answers is not in the document]
Comment (dlobato) [reply actions/copy_file:169]: apparently "Expression fragments inside f-strings cannot include backslashes"

**Review by mharista — APPROVED** [review]
LGTM

**Pull request merged**